Formula/gismo.rb

```diff
@@ -0,0 +1,65 @@
+class Gismo < Formula
+  desc "C++ library for isogeometric analysis (IGA)"
+  homepage "https://gismo.github.io"
+  url "https://github.com/gismo/gismo/archive/refs/tags/v21.12.0.tar.gz"
+  sha256 "4001b4c49661ca8b71baf915e773341e115d154077bef218433a3c1d72ee4f0c"
+  license "MPL-2.0"
+
+  head do
+    url "https://github.com/gismo/gismo.git", branch: "stable"
+  end
+
+  depends_on "cmake" => :build
+  depends_on "openblas"
+  depends_on "suite-sparse"
+  depends_on "superlu"
+
+  on_macos do
+    depends_on "libomp"
+  end
+
+  def install
+    # Set hardware-optimized compiler flags
+    target_arch =
+      case Hardware.oldest_cpu
+      when :arm_vortex_tempest then "apple-m1"
```

Comment:
What language/compiler is this being passed to?

Reply:
The code is written in C++ and we use CMake and try to support as many compilers as possible. We have our own [CMake macro](https://github.com/gismo/gismo/blob/stable/cmake/OptimizeForArchitecture.cmake) to determine the optimal compiler flags for the CPU/OS/compiler combo. There, `"apple-m1"` is the identifier for the first generation of Apple Silicon (https://github.com/gismo/gismo/blob/87b3db9f245f527a0504bf062892ab7e3e19e1b3/cmake/ofa/AutodetectArm.cmake#L377). Based on this identifier, we set compiler flags (https://github.com/gismo/gismo/blob/87b3db9f245f527a0504bf062892ab7e3e19e1b3/cmake/ofa/HandleArmOptions.cmake#L738).

Reply:
Note Homebrew will always use its own compiler flags whenever possible so I'm fairly sure this is doing nothing as-is and can be removed. @carlocab can confirm.

Reply:
See below